src/mining_minigame.c

```diff
@@ -0,0 +1,3371 @@
+            Terrain_UpdateLayerTileOnScreen(ptr, -1, 1);
+
+        if (sMiningUiState->cursorX != 11 && sMiningUiState->cursorY != 2) 
+            Terrain_UpdateLayerTileOnScreen(ptr, 1, -1);
+
+        if (sMiningUiState->cursorX != 0 && sMiningUiState->cursorY != 2) 
+            Terrain_UpdateLayerTileOnScreen(ptr, -1, -1);
+
+        if (sMiningUiState->layerMap[pos] != 6) 
+            Terrain_Pickaxe_OverwriteTiles(ptr);
+    }
+}
+
+static void Mining_UpdateTerrain(void) 
+{
+    u16* ptr = GetBgTilemapBuffer(2);
+    switch (sMiningUiState->tool) 
+    {
+        case RED_BUTTON:
+            Terrain_Hammer_OverwriteTiles(ptr);
+            break;
+        case BLUE_BUTTON:
+            Terrain_Pickaxe_OverwriteTiles(ptr);
+            break;
+    }
+}
+
+static void Task_MiningFadeAndExitMenu(u8 taskId) 
+{
+    if (!gPaletteFade.active) 
+    {
+        SetMainCallback2(sMiningUiState->leavingCallback);
+        Mining_FreeResources();
+        DestroyTask(taskId);
+    }
+}
+
+static void Mining_FreeResources(void) 
+{
+    if (sMiningUiState != NULL)
+        Free(sMiningUiState);
+
+    FreeAllWindowBuffers();
+    ResetSpriteData();
+    SetGpuReg(REG_OFFSET_WIN0H, 0);
+    SetGpuReg(REG_OFFSET_WIN0V, 0);
+    SetGpuReg(REG_OFFSET_WIN1H, 0);
+    SetGpuReg(REG_OFFSET_WIN1V, 0);
+    SetGpuReg(REG_OFFSET_WININ, 0);
+    SetGpuReg(REG_OFFSET_WINOUT, 0);
+}
+
+static void InitMiningWindows(void) 
+{
+    if (InitWindows(sWindowTemplates))
+    {
+        DeactivateAllTextPrinters();
+        ScheduleBgCopyTilemapToVram(0);
+#if FLAG_USE_DEFAULT_MESSAGE_BOX == FALSE
+        LoadBgTiles(GetWindowAttribute(WIN_MSG, WINDOW_BG), gMiningMessageBoxGfx, 0x1C0, 20);
+        LoadPalette(gMiningMessageBoxPal, BG_PLTT_ID(15), PLTT_SIZE_4BPP);
+        LoadPalette(gMiningMessageBoxPal, BG_PLTT_ID(14), PLTT_SIZE_4BPP);
+#elif FLAG_USE_DEFAULT_MESSAGE_BOX == TRUE
+        LoadBgTiles(GetWindowAttribute(WIN_MSG, WINDOW_BG), gMessageBox_Gfx, 0x1C0, 20);
+        LoadPalette(GetOverworldTextboxPalettePtr(), BG_PLTT_ID(15), PLTT_SIZE_4BPP);
+        Menu_LoadStdPalAt(BG_PLTT_ID(14));
+#endif
+        PutWindowTilemap(WIN_MSG);
+        CopyWindowToVram(WIN_MSG, COPYWIN_FULL);
+    }
+}
+
+static void PrintMessage(const u8 *string) 
+{
+    u32 letterSpacing = 0;
+    u32 x = 0;
+    u32 y = 1;
+
+    u8 txtColor[]= {TEXT_COLOR_WHITE, TEXT_COLOR_DARK_GRAY, TEXT_COLOR_LIGHT_GRAY};
+
+    DrawDialogFrameWithCustomTileAndPalette(WIN_MSG, FALSE, 20, 15);
+    FillWindowPixelBuffer(WIN_MSG, PIXEL_FILL(TEXT_COLOR_WHITE));
+    CopyWindowToVram(WIN_MSG, 3);
+    PutWindowTilemap(WIN_MSG);
+    AddTextPrinterParameterized4(WIN_MSG, FONT_NORMAL, x, y, letterSpacing, 1, txtColor, GetPlayerTextSpeedDelay(),string);
+    RunTextPrinters();
+}
+
+static u32 GetStressLevelPosition(void) 
+{
+    return sMiningUiState->stressLevelPos;
+}
+
+static bool32 IsStressLevelMax(void) 
+{
+    return GetStressLevelPosition() == STRESS_LEVEL_POS_MAX;
+}
```

Comment:
```suggestion
static bool32 IsStressLevelMax(void) 
{
    return sMiningUiState->stressLevelPos == STRESS_LEVEL_POS_MAX;
}
```
`GetStressLevelPosition` seems a bit redundant.

Reply:
Yeah I believe that was leftover code from psf
blame him >:)